### DISCUSSION TITLE
Help with HTMLTable adjustements

### DISCUSSION THREAD

**Discussion by amanfromfrance** [EvotecIT/PSWriteHTML#293] [Q&A] [answered]
Hello,

I am still improving my reporting files and have troubles to set the HTMLTables options to fit my requirements.
For example, with this code:

```powershell
New-HTMLSection -Invisible {
	New-HTMLTable -DataTable $emeaExpresswayC2ResourceUsage -ExcludeProperty peer1, peer2, registeredVideoCalls, registeredAudioCalls, rmsVideoCalls, rmsAudioCalls -Title 'Resource Consumption details' -HideFooter -DisableButtons -DisablePaging -DisableSearch -AutoSize -DisableAutoWidthOptimization {
		New-HTMLTableStyle -TextAlign center
		New-TableHeader -Color White -Title 'EMEA C2 Resource Consumption details' -Alignment center -FontSize 12 -FontWeight bold -BackGroundColor "#000000"
		New-TableHeader -Names peer1RegisteredVideoCalls,peer1RegisteredAudioCalls,peer1RmsVideoCalls,peer1RmsAudioCalls -Title $emeaExpresswayC2ResourceUsage.peer1[0] -Color White -Alignment center -BackGroundColor "#8f8f8f"
		New-TableHeader -Names peer2RegisteredVideoCalls,peer2RegisteredAudioCalls,peer2RmsVideoCalls,peer2RmsAudioCalls -Title $emeaExpresswayC2ResourceUsage.peer2[0] -Color White -Alignment center -BackGroundColor "#8f8f8f"
		New-TableHeader -Names 'date' -Title "" -Color White -Alignment center -BackGroundColor "#8f8f8f"
	}
}
```

That generates the table below:

![2021-11-22_110903](https://user-images.githubusercontent.com/58064665/142842727-6a2934a0-dea0-4186-94ba-fa3f8c9313b0.png)

I have 2 minor issues:

1. How can I change the columns labels? Those are retrieved from text files where I need them to be that specific to differentiate the data, however I'd like to display shorter versions in the table for clarity's sake (for example display "_**Reg. Video Calls**_" instead of "_**peer1RegisteredVideoCalls**_").
I searched in the examples and source files but I cannot find a way to do this.
2. How can I force the table to automatically resize to the page width and not display the ![2021-11-22_111456](https://user-images.githubusercontent.com/58064665/142843425-6752d876-cb46-470b-832c-5e387248da76.png) button to display hidden columns? ![2021-11-22_111651](https://user-images.githubusercontent.com/58064665/142843757-d84cf163-f6f5-4c8c-afe0-feecedfc6185.png)

Thanks in advance for your help!

**Comment by PrzemyslawKlys** [comment, marked as answer]
To be honest:
1. I used to believe I added this feature but maybe I just thought I did. New-TableHeader -Name 'xxx' -Title 'yyy' may work - if it doesn't work - that means I have not added it and it should be added - for example with a new parameter. Need to think about how to do it properly. For now, you can probably read CSV and convert CSV to [PSCustomObjects] and then push it to the table, already prepared with names. Maybe `New-TableContent` would work, but i doubt it touches headers.
2. DisableResponsive switch will disable +. You can try other options such as scroll x or wordbreak to get it to act differently.

**Reply by amanfromfrance** [reply]
Thanks for your feedback!

1. I tried to add the **_New-TableHeader -Name 'xxx' -Title 'yyy_** command on some column names but it changed nothing.
    It would have been great to be able to do it, but I can leave with the column headers not being user friendly though.
2. I've tried 2 solutions:
    - Using the "**DisableResponsiveTable**" parameter:
      The table is not automatically resized and its width is fixed & cropped:
      ![2021-11-23_121056](https://user-images.githubusercontent.com/58064665/143015463-ffd03398-237f-44c6-a06e-2de7a97fc5c9.png)
    - Using the "**ScrollX**" parameter:
      It is not exactly my expected result but at least it is an improvement:
      ![2021-11-23_121156](https://user-images.githubusercontent.com/58064665/143015931-cf0c3f65-a384-4889-8924-0433ca8d36ce.png)

Below is the code used to generate both tables:

```powershell
New-HTMLTable -DataTable $emeaExpresswayC1ResourceUsage -DisableResponsiveTable -ExcludeProperty peer1, peer2, registeredVideoCalls, registeredAudioCalls, rmsVideoCalls, rmsAudioCalls -Title 'Resource Consumption details' -HideFooter -DisableButtons -DisablePaging -DisableSearch -AutoSize {
	New-HTMLTableStyle -TextAlign center
	New-TableHeader -Title 'Stats' -Color White -ColumnCount 2 -BackGroundColor White
	New-TableHeader -Title 'EMEA C1 Resource Consumption details' -Color White -ColumnCount 8 -Alignment center -FontSize 12 -FontWeight bold -BackGroundColor "#000000"
	New-TableHeader -Names peer2RegisteredVideoCalls,peer2RegisteredAudioCalls,peer2RmsVideoCalls,peer2RmsAudioCalls -Title $emeaExpresswayC1ResourceUsage.peer2[0] -Color White -Alignment center -BackGroundColor "#8f8f8f"
	New-TableHeader -Names peer1RegisteredVideoCalls,peer1RegisteredAudioCalls,peer1RmsVideoCalls,peer1RmsAudioCalls -Title $emeaExpresswayC1ResourceUsage.peer1[0] -Color White -Alignment center -BackGroundColor "#8f8f8f"
	New-TableHeader -Names 'date' -Title "" -Color White -Alignment center -BackGroundColor "#8f8f8f"
	New-TableHeader -Names 'rmsLicensesUsage' -Title '' -Color White -Alignment center -BackGroundColor "#8f8f8f"
	New-HTMLTableCondition -Name 'date' -ComparisonType string -Operator contains -Value " " -FontWeight bold -Color "#50BE87" -HighlightHeaders 'date'
	New-HTMLTableCondition -Name 'rmsLicensesUsage' -ComparisonType string -Operator contains -Value "%" -FontWeight bold -Color "#4BB4E6" -HighlightHeaders 'rmsLicensesUsage'
} -FreezeColumnsLeft 1

New-HTMLTable -DataTable $emeaExpresswayC2ResourceUsage -ScrollX -ExcludeProperty peer1, peer2, registeredVideoCalls, registeredAudioCalls, rmsVideoCalls, rmsAudioCalls -Title 'Resource Consumption details' -HideFooter -DisableButtons -DisablePaging -DisableSearch -AutoSize -DisableAutoWidthOptimization {
	New-HTMLTableStyle -TextAlign center
	New-TableHeader -Title 'Stats' -Color White -ColumnCount 2 -BackGroundColor White
	New-TableHeader -Title 'EMEA C2 Resource Consumption details' -Color White -ColumnCount 8 -Alignment center -FontSize 12 -FontWeight bold -BackGroundColor "#000000"
	New-TableHeader -Names peer1RegisteredVideoCalls,peer1RegisteredAudioCalls,peer1RmsVideoCalls,peer1RmsAudioCalls -Title $emeaExpresswayC2ResourceUsage.peer1[0] -Color White -Alignment center -BackGroundColor "#8f8f8f"
	New-TableHeader -Names peer2RegisteredVideoCalls,peer2RegisteredAudioCalls,peer2RmsVideoCalls,peer2RmsAudioCalls -Title $emeaExpresswayC2ResourceUsage.peer2[0] -Color White -Alignment center -BackGroundColor "#8f8f8f"
	New-TableHeader -Names 'date' -Title "" -Color White -Alignment center -BackGroundColor "#8f8f8f"
	New-TableHeader -Names 'rmsLicensesUsage' -Title '' -Color White -Alignment center -BackGroundColor "#8f8f8f"
	New-HTMLTableCondition -Name 'date' -ComparisonType string -Operator contains -Value " " -FontWeight bold -Color "#50BE87" -HighlightHeaders 'date'
	New-HTMLTableCondition -Name 'rmsLicensesUsage' -ComparisonType string -Operator contains -Value "%" -FontWeight bold -Color "#4BB4E6" -HighlightHeaders 'rmsLicensesUsage'
}
```
